Apps/WebClient/src/NewClientApp/src/utility/promiseUtility.ts

```diff
@@ -31,4 +34,26 @@ export default abstract class PromiseUtility {
     public static delay(milliseconds: number): Promise<unknown> {
         return new Promise((resolve) => setTimeout(resolve, milliseconds));
     }
+
+    /**
+     * Wraps a promise so a loader is enabled beforehand and disabled afterward.
+     * @param promise The provided Promise.
+     * @param loader A loader to invoke.
+     * @param reason A reason for invoking the loader.
+     * @returns A new Promise.
+     */
+    public static async withLoader<T>(
```

Comment:
I would argue this could just be added directly to the loading store. Putting it here splits the tracking logic need to import 3 items in components that might both listen and set loading statuses. Moving this to the loadingStore reduces the complexity of the implementation steps.

```Typescript
loadingStore.trackPromise<T>(promise: Promise<T>, loader: Loader, reason?:string);

loadingStore.trackPromise(...yourPromiseFunctionInvocation, loader, "fancy reason");

```
Then, catch and finally works very similar to try catches and you can chain as many finally calls as needed. So you could rely on the promise's finally rather than a try catch, it will trigger regardless of promise rejections.
```Typescript
// This may need some playing but your tracking method body could be simplified to:
setIsLoading(true, loader, reason);
promise.finally(() => setIsLoading(false, loader, reason);
```

Reply:
Done.